### PR TITLE
fix(sql) update files from 21.10.0 to 22.04.0

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Add missing upgrade scripts for 21.10.x in develop

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by AssanDiomande — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: www/install/php/Update-21.10.2.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    if (!$pearDB->isColumnExist('contact', 'contact_js_effects')) {
+        $pearDB->query(
+            "ALTER TABLE `contact`
+            ADD COLUMN `contact_js_effects` enum('0','1') DEFAULT '0'
+            AFTER `contact_comment`"
+        );
+    }
+
+    $errorMessage = 'Unable to delete logger entry in cb_tag';
```
Comment (a-launois) [inline www/install/php/Update-21.10.2.php:42]: Delete this part

File: www/install/php/Update-21.10.2.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    $statement = $pearDB->query("DELETE FROM cb_tag WHERE tagname = 'logger'");
+    $errorMessage = 'Unable to update the description in cb_field';
+    $statement = $pearDB->query("
+        UPDATE cb_field
+        SET `description` = 'Time in seconds to wait between each connection attempt (Default value: 30s).'
+        WHERE `cb_field_id` = 31
+    ");
```
Comment (a-launois) [inline www/install/php/Update-21.10.2.php:49]: Delete

File: www/install/php/Update-21.10.4.php [hunk 3]
```diff
@@ -19,11 +19,52 @@
...
+ * Query with transaction
+ */
+try {
+    $pearDB->beginTransaction();
+    $errorMessage = 'Unable to delete logger entry in cb_tag';
+    $statement = $pearDB->query("DELETE FROM cb_tag WHERE tagname = 'logger'");
+    $errorMessage = 'Unable to update the description in cb_field';
+    $statement = $pearDB->query("
+        UPDATE cb_field
+        SET `description` = 'Time in seconds to wait between each connection attempt. The default value is 30s.'
```
Comment (a-launois) [inline www/install/php/Update-21.10.4.php:39]: Time in seconds to wait between each connection attempt (Default value: 30s).
```suggestion
        SET `description` = 'Time in seconds to wait between each connection attempt (Default value: 30s).'
```

File: www/install/php/Update-21.10.4.php [hunk 4]
```diff
@@ -19,11 +19,52 @@
...
+        WHERE `cb_field_id` = 31
+    ");
+
+    $errorMessage  = 'Unable to delete logger entry in cb_tag';
+    $statement = $pearDB->query("DELETE FROM cb_tag WHERE tagname = 'logger'");
```
Comment (a-launois) [inline www/install/php/Update-21.10.4.php:44]:
```suggestion
```

**Review by AssanDiomande — DISMISSED** [review]

File: www/install/php/Update-21.10.2.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    $errorMessage = 'Impossible to add "contact_js_effects" column to "contact" table';
+
+    if (!$pearDB->isColumnExist('contact', 'contact_js_effects')) {
+        $pearDB->query(
+            "ALTER TABLE `contact`
+            ADD COLUMN `contact_js_effects` enum('0','1') DEFAULT '0'
+            AFTER `contact_comment`"
+        );
+    }
+
```
Comment (kduret) [inline www/install/php/Update-21.10.2.php:41]:
```suggestion
```

**Review by kduret — DISMISSED** [review]

File: www/install/php/Update-21.10.2.php (hunk 5 above)
Comment (a-launois) [inline www/install/php/Update-21.10.2.php]:
```suggestion
```

File: www/install/php/Update-21.10.4.php [hunk 6]
```diff
@@ -19,11 +19,50 @@
...
+    $errorMessage = 'Unable to delete logger entry in cb_tag';
+    $statement = $pearDB->query("DELETE FROM cb_tag WHERE tagname = 'logger'");
+    $errorMessage = 'Unable to update the description in cb_field';
+    $statement = $pearDB->query("
+        UPDATE cb_field
+        SET `description` = 'Time in seconds to wait between each connection attempt (Default value: 30s).'
+        WHERE `cb_field_id` = 31
+    ");
+
+    $pearDB->commit();
```
Comment (callapa) [inline www/install/php/Update-21.10.4.php:43]: As there was only one rollback, you only need to use one commit and one transaction.
```suggestion
```

File: www/install/php/Update-21.10.4.php [hunk 7]
```diff
@@ -19,11 +19,50 @@
...
+
+    $errorMessage = 'Cannot purge host macros';
+    $cache = loadHosts($pearDB);
+    $pearDB->beginTransaction();
```
Comment (callapa) [inline www/install/php/Update-21.10.4.php:47]:
```suggestion
```

**Review by callapa — DISMISSED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by AssanDiomande — APPROVED** [review]

**Pull request merged**